indexing-service/src/main/java/io/druid/indexing/overlord/PortFinder.java

```diff
@@ -27,16 +27,19 @@
 import java.io.IOException;
 import java.net.ServerSocket;
 import java.net.SocketException;
+import java.util.List;
 import java.util.Set;
 
 public class PortFinder
 {
   private final Set<Integer> usedPorts = Sets.newHashSet();
   private final int startPort;
+  private final List<Integer> candidatePorts;
 
-  public PortFinder(int startPort)
+  public PortFinder(int startPort, List<Integer> candidatePorts)
```

Comment:
Instead of modifying the existing constructor, you could have a separate constructor with both the properties and to be used only when `candidatePorts` is non-empty.

Reply:
I am not very willing to do so for I have to add an extra `if statement` which will make code a little ugly than before....  If more reviewers agree with you, then it will be fine and I'll modify this.